### PR TITLE
#211 - Fees Improvement part2 

### PR DESCRIPTION
**Pull request by quent043**
# New PR:

## Useful info

- Trello: https://trello.com/c/z3pJob2Y/211-fees-improvement
- Description: Added 2 new fees

## Auto-Review checklist

- [ ] Check if there is no merge conflict
- [ ] If you have new .env variable, check if you add it in the .env.example file

## Typing

- [ ] Check solhint feedbacks: `npm run solhint`

### CODE REVIEW

File: contracts/ServiceRegistry.sol [hunk 1]
```diff
@@ -226,7 +226,9 @@ contract ServiceRegistry is Initializable, ERC2771RecipientUpgradeable, UUPSUpgr
      * @param _platformId platform ID on which the Service token was minted
      * @param _serviceDataUri token Id to IPFS URI mapping
      */
-    function createOpenServiceFromBuyer(uint256 _platformId, string calldata _serviceDataUri) public returns (uint256) {
+    function createOpenServiceFromBuyer(uint256 _platformId, string calldata _serviceDataUri) public payable returns (uint256) {
+        uint16 servicePostingFee = talentLayerPlatformIdContract.getServicePostingFee(_platformId);
+        require(msg.value == servicePostingFee, "Non-matching funds");
         talentLayerPlatformIdContract.isValid(_platformId);
```
Comment (0xRomain) [inline contracts/ServiceRegistry.sol:232]: now we got two external calls here, that's not optimal, we can probably do the same with only one

File: contracts/TalentLayerPlatformID.sol [hunk 2]
```diff
@@ -267,6 +291,30 @@ contract TalentLayerPlatformID is ERC721Upgradeable, AccessControlUpgradeable, U
         emit ArbitrationFeeTimeoutUpdated(_platformId, _arbitrationFeeTimeout);
     }
 
+    /**
+     * @notice Allows a platform to update the service posting fee for the platform
+     * @param _platformId The platform id
+     * @param _servicePostingFee The new fee
+     */
+    function updateServicePostingFee(uint256 _platformId, uint16 _servicePostingFee) public {
```
Comment (0xRomain) [inline contracts/TalentLayerPlatformID.sol:299]: problem here, the fee is in eth so uint256

File: contracts/TalentLayerPlatformID.sol [hunk 3]
```diff
@@ -267,6 +291,30 @@ contract TalentLayerPlatformID is ERC721Upgradeable, AccessControlUpgradeable, U
...
+        platforms[_platformId].servicePostingFee = _servicePostingFee;
+        emit ServicePostingFeeUpdated(_platformId, _servicePostingFee);
+    }
+
+    /**
+     * @notice Allows a platform to update the proposal posting fee for the platform
+     * @param _platformId The platform id
+     * @param _proposalPostingFee The new fee
+     */
+    function updateProposalPostingFee(uint256 _platformId, uint16 _proposalPostingFee) public {
```
Comment (0xRomain) [inline contracts/TalentLayerPlatformID.sol:311]: same issue

File: contracts/interfaces/ITalentLayerPlatformID.sol [hunk 4]
```diff
@@ -60,6 +62,14 @@ interface ITalentLayerPlatformID is IERC721Upgradeable {
 
     function updateMinArbitrationFeeTimeout(uint256 _minArbitrationFeeTimeout) external;
 
+    function getServicePostingFee(uint256 _platformId) external view returns (uint16);
+
+    function getProposalPostingFee(uint256 _platformId) external view returns (uint16);
+
+    function updateServicePostingFee(uint256 _platformId, uint16 _servicePostingFee) external;
```
Comment (0xRomain) [inline contracts/interfaces/ITalentLayerPlatformID.sol:69]: typing to update everywhere with uint256

File: contracts/ServiceRegistry.sol (hunk 1 above)
Comment (0xRomain) [reply contracts/ServiceRegistry.sol:232]: we can do a global getPlatform probably, can you check the gas impact please (now and after getPlatform implemented)

Comment (quent043) [reply contracts/ServiceRegistry.sol:232]: Actually "isValid" check is already done  in getServicePostingFee
So can be removed

**Pull request merged**